code/core/src/csf-tools/ConfigFile.test.ts

```diff
@@ -1462,6 +1462,318 @@ describe('ConfigFile', () => {
+      `;
+
+      const config = loadConfig(source).parse();
+      config.removeImport(['dirname'], 'node:path');
+
+      const parsed = babelPrint(config._ast);
+
+      expect(parsed).toMatchInlineSnapshot(`
+        const config: StorybookConfig = { };
+        export default config;
+      `);
+    });
+
+    it(`does nothing when require import does not exist`, () => {
+      const source = dedent`
+        const config: StorybookConfig = { };
+        export default config;
+      `;
+
+      const config = loadConfig(source).parse();
+      config.removeImport('path', 'path');
+
+      const parsed = babelPrint(config._ast);
+
+      expect(parsed).toMatchInlineSnapshot(source);
+    });
+
+    it(`does nothing when trying to remove non-existent named require import`, () => {
+      const source = dedent`
+        const { dirname } = require('path');
+        const config: StorybookConfig = { };
+        export default config;
+      `;
+
+      const config = loadConfig(source).parse();
+      config.removeImport(['nonexistent'], 'path');
+
+      const parsed = babelPrint(config._ast);
+
+      expect(parsed).toMatchInlineSnapshot(source);
+    });
+
+    it(`does nothing when trying to remove non-existent named ES6 import`, () => {
+      const source = dedent`
+        import { dirname } from 'path';
+        const config: StorybookConfig = { };
+        export default config;
+      `;
+
+      const config = loadConfig(source).parse();
+      config.removeImport(['nonexistent'], 'path');
+
+      const parsed = babelPrint(config._ast);
+
+      expect(parsed).toMatchInlineSnapshot(source);
+    });
+
+    it(`removes ES6 import while preserving require import`, () => {
+      const source = dedent`
+        import { readFile } from 'fs';
+        const { dirname } = require('path');
+        const config: StorybookConfig = { };
+        export default config;
+      `;
+
+      const config = loadConfig(source).parse();
+      config.removeImport(['readFile'], 'fs');
+
+      const parsed = babelPrint(config._ast);
+
+      expect(parsed).toMatchInlineSnapshot(`
+        const { dirname } = require('path');
+        const config: StorybookConfig = { };
+        export default config;
+      `);
+    });
+
+    it(`removes require import while preserving ES6 import`, () => {
+      const source = dedent`
+        import { readFile } from 'fs';
+        const { dirname } = require('path');
+        const config: StorybookConfig = { };
+        export default config;
+      `;
+
+      const config = loadConfig(source).parse();
+      config.removeImport(['dirname'], 'path');
+
+      const parsed = babelPrint(config._ast);
+
+      expect(parsed).toMatchInlineSnapshot(`
+        import { readFile } from 'fs';
+        const config: StorybookConfig = { };
+        export default config;
+      `);
+    });
+
+    // TODO: dual default import+require
+    // TODO: preserve existing non-matching imports for all use cases
```

Comment:
ugh, missed those.

* [ ] Add tests

Reply:
Looked back at this, I'm satisfied with the amount of tests already done. I think those TODOs can just be removed